src/software/proto/message_translation/ssl_referee_test.cpp

```diff
@@ -2,4 +2,157 @@
 
 #include <gtest/gtest.h>
 
-// TODO (Issue #1458): Add tests
+#include "software/proto/ssl_referee.pb.h"
+#include "software/sensor_fusion/refbox_data.h"
+#include "software/world/team_colour.h"
+
+class RefboxStageTest
+    : public ::testing::TestWithParam<std::tuple<RefboxStage, SSL_Referee_Stage>>
+{
+};
+
+TEST_P(RefboxStageTest, test_refbox_stage)
+{
+    RefboxStage expected    = std::get<0>(GetParam());
+    SSL_Referee_Stage input = std::get<1>(GetParam());
+
+    SSL_Referee ref;
+    ref.set_stage(input);
+
+    ASSERT_EQ(expected, createRefboxStage(ref));
+}
+
+INSTANTIATE_TEST_CASE_P(
+    RefboxStageTests, RefboxStageTest,
+    ::testing::Values(
+        std::make_tuple(RefboxStage::NORMAL_FIRST_HALF_PRE,
+                        SSL_Referee_Stage_NORMAL_FIRST_HALF_PRE),
+        std::make_tuple(RefboxStage::NORMAL_FIRST_HALF,
+                        SSL_Referee_Stage_NORMAL_FIRST_HALF),
+        std::make_tuple(RefboxStage::NORMAL_HALF_TIME,
+                        SSL_Referee_Stage_NORMAL_HALF_TIME),
+        std::make_tuple(RefboxStage::NORMAL_SECOND_HALF_PRE,
+                        SSL_Referee_Stage_NORMAL_SECOND_HALF_PRE),
+        std::make_tuple(RefboxStage::NORMAL_SECOND_HALF,
+                        SSL_Referee_Stage_NORMAL_SECOND_HALF),
+        std::make_tuple(RefboxStage::EXTRA_TIME_BREAK,
+                        SSL_Referee_Stage_EXTRA_TIME_BREAK),
+        std::make_tuple(RefboxStage::EXTRA_FIRST_HALF_PRE,
+                        SSL_Referee_Stage_EXTRA_FIRST_HALF_PRE),
+        std::make_tuple(RefboxStage::EXTRA_FIRST_HALF,
+                        SSL_Referee_Stage_EXTRA_FIRST_HALF),
+        std::make_tuple(RefboxStage::EXTRA_HALF_TIME, SSL_Referee_Stage_EXTRA_HALF_TIME),
+        std::make_tuple(RefboxStage::EXTRA_SECOND_HALF_PRE,
+                        SSL_Referee_Stage_EXTRA_SECOND_HALF_PRE),
+        std::make_tuple(RefboxStage::EXTRA_SECOND_HALF,
+                        SSL_Referee_Stage_EXTRA_SECOND_HALF),
+        std::make_tuple(RefboxStage::PENALTY_SHOOTOUT_BREAK,
+                        SSL_Referee_Stage_PENALTY_SHOOTOUT_BREAK),
+        std::make_tuple(RefboxStage::PENALTY_SHOOTOUT,
+                        SSL_Referee_Stage_PENALTY_SHOOTOUT),
+        std::make_tuple(RefboxStage::POST_GAME, SSL_Referee_Stage_POST_GAME)));
+
+class RefboxGameStateTest
+    : public ::testing::TestWithParam<
+          std::tuple<RefboxGameState, SSL_Referee_Command, TeamColour>>
+{
+};
+
+TEST_P(RefboxGameStateTest, test_refbox_game_state)
+{
+    RefboxGameState expected  = std::get<0>(GetParam());
+    SSL_Referee_Command input = std::get<1>(GetParam());
+    TeamColour team_colour    = std::get<2>(GetParam());
+
+    SSL_Referee ref;
+    ref.set_command(input);
+
+    ASSERT_EQ(expected, createRefboxGameState(ref, team_colour));
```

Comment:
is a dynamic param somewhere setting that we treat Yellow as friendly?

Reply:
no, team_colour is the friendly colour